### DISCUSSION TITLE
GUI threads limited to 40? (even if computer supports more?)

### DISCUSSION THREAD

**Discussion by JM-Bader** [vdemichev/DiaNN#390] [Q&A]
I would like to set the threads of DIANN to the number of supported hyperthreads of my PC, which is 160. 

However, the GUI field that is set to 40 by default also appears to be capped at 40. Setting a higher number there will result in 40. 
![image](https://user-images.githubusercontent.com/98888664/169849719-d08216d5-3386-4c3a-a089-273963ab0647.png)

Moreover, when using the --threads command with a number higher than 40 (while the GUI field is at 40) also results in 40 (as by the log feed shown). 

Further, DIANN log feed says 'logical CPU cores: 40'.
![image](https://user-images.githubusercontent.com/98888664/169852240-64755dfc-183c-4263-bac5-9deeaf217407.png)

However, the computer has 80 cores and 160 logical processors:
![image](https://user-images.githubusercontent.com/98888664/169852425-eb24c28d-13ff-4eb6-9a4b-9ff42761e231.png)

Q1: How do increase the number of threads of DIANN beyond 40?

**Comment by vdemichev** [comment]
Please use the --threads command

**Reply by JM-Bader** [reply]
I do use it, see above. I have set it to 160. DIANN overwrites it to 40 (see second screenshot)

**Reply by vdemichev** [reply]
OK, I see, I will need to check the code to see why this happens. Please keep using 40 meanwhile. Can run multiple instances of DIA-NN on the same PC though.

**Reply by JM-Bader** [reply]
My observation is that DIANN misdiagnoses the number of logical cores on the computer. Maybe DIANN has a sanity check to limit the number of threads to the (misdiagnosed) number of cores. Could that be?

**Reply by JM-Bader** [reply]
With other search engines, splitting the search can be an issue that causes i) different protein grouping, ii) batch effects. Thus I am a bit skeptical that this is a good idea in DIANN. Is there documentation about this / where?

**Reply by JM-Bader** [reply]
If there can be a quick fix to allow for higher thread numbers, I would be very happy to test such a version.

**Reply by vdemichev** [reply]
Yes, here https://github.com/bigbio/quantms/issues/164

**Reply by JM-Bader** [reply]
> OK, I see, I will need to check the code to see why this happens. Please keep using 40 meanwhile. Can run multiple instances of DIA-NN on the same PC though.

Hey Vadim, I know you have a lot to do and appreciate all the swift support you provide here.  Is there a chance you have had a look at this? Do you think there could be a fix/version soon which permits more than 40 threads to be used?

**Reply by JM-Bader** [reply]
My observation now is that it is processor-dependent. 
With some processors, DIANN actually identifies the correct number of physical and logical cores and permits to set the threads to the number of logical cores: 

example of correct cores (DIANN sets it to 32 = physical cores by default, but permits up to 64 = logical cores)
![image](https://user-images.githubusercontent.com/98888664/175353213-a5332f26-f018-4a67-b0f1-92542e518dbe.png)
![image](https://user-images.githubusercontent.com/98888664/175353316-bc616f54-baef-43cf-a67a-dc45635bc842.png)
![image](https://user-images.githubusercontent.com/98888664/175353407-69c6c4e6-efe5-446f-b8c9-e600ccf572f6.png)

But I haven't identified a pattern yet. 
It appears that DIANN has problems reading out the number of cores from some processors.

**Comment by JM-Bader** [comment]
I now found cases in which DIANN identifies a higher number of cores than it lets you set it. 
In conclusion, it is not only a "diagnostic issue" of DIANN

This example: 
processor: 64 physcial cores, 128 logical cores
DIANN recognizes: 64 logical cores
DIANN lets on set in GUI field: 64 threads
DIANN overwrites in log: "Thread number set to 40"

DIANN
![image](https://user-images.githubusercontent.com/98888664/175510436-a3a2e021-eab3-4054-9f0e-fcf61971f879.png)

processor
![image](https://user-images.githubusercontent.com/98888664/175510679-4a9d064a-54f8-4035-a190-b0aab4e69ce1.png)




![image](https://user-images.githubusercontent.com/98888664/175510561-b40b356d-ffcf-42d0-abe9-98c8b686f495.png)
